xbmc/dialogs/GUIDialogProgress.cpp

```diff
@@ -176,7 +176,12 @@ void CGUIDialogProgress::ShowProgressBar(bool bOnOff)
 bool CGUIDialogProgress::Wait(int progresstime /*= 10*/)
 {
   CEvent m_done;
-  while (!m_done.WaitMSec(progresstime) && m_active && !m_bCanceled)
+  return WaitOnEvent(m_done, progresstime);
+}
+
+bool CGUIDialogProgress::WaitOnEvent(CEvent& event, int progresstime)
+{
+  while (!event.WaitMSec(progresstime) && m_active && !m_bCanceled)
```

Comment:
I found another place where complexity was shoved into a single line instead of spread out over multiple lines. This could be:

```c++
while (!event.WaitMSec(progresstime))
{
  if (!m_active)
    break;

  if (m_bCanceled)
    break;

  Progress();
}
```

It adds a lot of lines, but see how it's more readable and understandable? Also important is that these conditions can change -- having checks on different lines means we can modify any, and not affect the git history of the remaining ones.